src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/DownloadVideo/index.tsx

```diff
@@ -78,13 +103,59 @@ export const DownloadVideo = () => {
     value: string;
   }>(options[options.length - 1]);
 
+  const [toggleAllowDownload, setToggleAllowDownload] = useState(
+    video.show_download,
+  );
```

Comment:
Shouldn't you have a useEffect listening to `video.show_download` value change in case some other user change it on his side and the video is updated through the websocket ?